lib/improver/tests/wind_calculations/wind_components/test_ResolveWindComponents.py

```diff
@@ -0,0 +1,230 @@
+    """Tests the resolve_wind_components method"""
+
+    def setUp(self):
+        """Set up some arrays to convert"""
+        self.plugin = ResolveWindComponents()
+        wind_speed = 10.*np.ones((4, 4), dtype=np.float32)
+        wind_angle = np.array([[0., 30., 45., 60.],
+                               [90., 120., 135., 150.],
+                               [180., 210., 225., 240.],
+                               [270., 300., 315., 330.]])
+        self.wind_cube = set_up_cube(wind_speed, "wind_speed", "knots")
+        self.directions = set_up_cube(
+            wind_angle, "wind_to_direction", "degrees")
+        self.adjustments = np.zeros((4, 4), dtype=np.float32)
+
+    def test_basic(self):
+        """Test function returns correct type"""
+        uspeed, vspeed = self.plugin.resolve_wind_components(
+            self.wind_cube, self.directions, self.adjustments)
+        self.assertIsInstance(uspeed, iris.cube.Cube)
+        self.assertIsInstance(vspeed, iris.cube.Cube)
+
+    def test_values(self):
+        """Test correct values are returned for well-behaved angles"""
+        expected_uspeed = 5.*np.array(
+            [[0., 1., np.sqrt(2.), np.sqrt(3.)],
+             [2., np.sqrt(3.), np.sqrt(2.), 1.],
+             [0., -1., -np.sqrt(2.), -np.sqrt(3.)],
+             [-2., -np.sqrt(3.), -np.sqrt(2.), -1.]])
+
+        expected_vspeed = 5*np.array(
+            [[2., np.sqrt(3.), np.sqrt(2.), 1.],
+             [0., -1., -np.sqrt(2.), -np.sqrt(3.)],
+             [-2., -np.sqrt(3.), -np.sqrt(2.), -1.],
+             [0., 1., np.sqrt(2.), np.sqrt(3.)]])
+
+        uspeed, vspeed = self.plugin.resolve_wind_components(
+            self.wind_cube, self.directions, self.adjustments)
+        self.assertArrayAlmostEqual(uspeed.data, expected_uspeed)
+        self.assertArrayAlmostEqual(vspeed.data, expected_vspeed)
+
+
+class Test_process(IrisTest):
+    """Tests the process method"""
+
+    def setUp(self):
+        """Create dummy cubes for tests"""
+        self.plugin = ResolveWindComponents()
+        wind_speed_data = np.array(
+            [[6, 5, 4, 3], [8, 6, 4, 4], [12, 8, 6, 5]], dtype=np.float32)
+        self.wind_speed_cube = set_up_cube(
+            wind_speed_data, "wind_speed", "knots")
+
+        wind_direction_data = np.array(
+            [[138, 142, 141, 141], [141, 143, 140, 142],
+             [142, 146, 141, 142]], dtype=np.float32)
+        self.wind_direction_cube = set_up_cube(
+            wind_direction_data, "wind_to_direction", "degrees")
+
+        self.expected_u = np.array([
+            [3.774919, 2.902824, 2.406847, 1.826062],
+            [4.683339, 3.386423, 2.456537, 2.374629],
+            [6.829466, 4.149148, 3.593584, 2.963243]], dtype=np.float32)
+
+        self.expected_v = np.array([
+            [-4.663688, -4.071071, -3.194854, -2.380230],
+            [-6.485857, -4.952993, -3.156806, -3.218872],
+            [-9.867035, -6.839924, -4.804805, -4.027306]], dtype=np.float32)
```

Comment:
Winds are from the south east, therefore U should be negative and v positive. These are the other way around.